### ISSUE TITLE
Deprecated Notices in WCPay 5.6.0

### ISSUE THREAD

**Issue by htdat** [Automattic/woocommerce-payments#5776]
### Describe the bug

I've just reviewed debug.log from the provided testing site by GlobalStep p1678813826134229/1678768526.480299-slack-C011ENB20Q1 (with the main purpose of checking errors from the platform project #5321) and found out these notices.

Not urgent but I think we will need to fix them eventually. 

> The Automattic\WooCommerce\Admin\API\Options::get_options function is deprecated since version 3.1.
> 
> woocommerce_blocks_checkout_order_processed is deprecated since version 7.2.0! Use woocommerce_store_api_checkout_order_processed instead. This action was deprecated in WooCommerce Blocks version 7.2.0. Please use woocommerce_store_api_checkout_order_processed instead.
> 
> The Automattic\WooCommerce\Admin\API\Options::update_options function is deprecated since version 3.1.
> 
> Payment gateway with handle 'WCPAY_BLOCKS_CHECKOUT' has been deactivated in Cart and Checkout blocks because its dependency 'wc-blocks-registry' is not registered. Read the docs about registering assets for payment methods: https://github.com/woocommerce/woocommerce-blocks/blob/060f63c04f0f34f645200b5d4da9212125c49177/docs/third-party-developers/extensibility/checkout-payment-methods/payment-method-integration.md#registering-assets
> 
> The Automattic\WooCommerce\Admin\API\Options::get_options function is deprecated since version 3.1.
> 

### To Reproduce

1. Go to '…'
2. Click on '…'
3. Scroll down to …'

### Desktop (please complete the following information):

* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

### Smartphone (please complete the following information):

* Device: [e.g. iPhone6]
* OS: [e.g. iOS8.1]
* Browser [e.g. stock browser, safari]
* Version [e.g. 22]

**Comment by Onyx808** [comment]
Site running PHP 8+ get the depricated and other warnings

```
( ! ) Deprecated: Return type of WCPay\Core\Server\Response::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wp-content/plugins/woocommerce-payments/includes/core/server/class-response.php on line 49
--
```

```
( ! ) Deprecated: Return type of WCPay\Core\Server\Response::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /wp-content/plugins/woocommerce-payments/includes/core/server/class-response.php on line 60
--
```

**Comment by nerrad** [comment]
Hey folks, this seems like an odd one to have been sitting with no fix for over a year. It's [surfacing for the community](https://woocommercecommunity.slack.com/archives/C03MCL9RTAR/p1719222876584869). (cc @vbelolapotkov for awareness).

**Comment by vbelolapotkov** [comment]
Thanks for the ping @nerrad , I'll try to bump its priority and assign to the best team.

**Comment by timur27** [comment]
Quick update after reviewing all the notices/errors, as it's been a while since they were discovered and things have changed in some areas:

* `offsetGet/offsetSet` notices occurring with PHP8 and mentioned [in the comments](https://github.com/Automattic/woocommerce-payments/issues/5776#issuecomment-1546147479) were fixed in [`5.9.0`](https://github.com/Automattic/woocommerce-payments/blob/develop/changelog.txt#L1066) by [this PR](https://github.com/Automattic/woocommerce-payments/pull/6050).

* `The Automattic\WooCommerce\Admin\API\Options::get_options/update_options function is deprecated since version 3.1.` notice was part of WooCommerce and was fixed by the following steps:
    * https://github.com/woocommerce/woocommerce/pull/32891 fixes the occurrence of the notice when opening `Payments -> Settings`.
    * https://github.com/woocommerce/woocommerce/pull/49092 (which was merged last week) fixes a similar notice but on the `Analytics -> Orders` page.

* `Payment gateway with handle 'WCPAY_BLOCKS_CHECKOUT...` notice cannot be reproduced anymore, but I could make it appear after commenting out [this asset registration step](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Blocks/AssetsController.php#L58) in WooCommerce.
    * The conclusion here is that the error has been fixed over time by WooCommerce.
    * I debugged locally and at [this point](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Blocks/Payments/Api.php#L144) `WCPAY_BLOCKS_CHECKOUT` indeed has `wc-blocks-registry` as a dependency, but it's passing the check and [this error](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Blocks/Payments/Api.php#L147) is not thrown.

* `woocommerce_blocks_checkout_order_processed is deprecated since version 7.2.0` is being fixed by [this PR](https://github.com/Automattic/woocommerce-payments/pull/9093).

Hence, all the bullet points except for the last one were fixed at this point. The last point will be fixed by https://github.com/Automattic/woocommerce-payments/pull/9093.
